### PR TITLE
[#1679] Integration tests for the mongo db based device registry

### PR DESCRIPTION
**Pull request by kaniyan**

### CODE REVIEW

File: tests/pom.xml [hunk 1]
```diff
@@ -223,6 +239,54 @@ Test cases are run against Docker images of Hono server + (Apache Qpid Dispatch
               <autoCreateCustomNetworks>true</autoCreateCustomNetworks>
               <startParallel>false</startParallel>
               <images>
+                <!-- ##### Mongo DB instance for the device registry ##### -->
+                <image>
+                  <name>${docker.image.org-name}/hono-mongodb-test</name>
+                  <alias>mongo</alias>
+                  <build>
+                    <imagePullPolicy>IfNotPresent</imagePullPolicy>
```
Comment (sophokles73) [inline tests/pom.xml:247]: IMHO we should also skip building the image if the MongoDB profile is disabled

Comment (kaniyan) [reply tests/pom.xml:253]: It makes sense and I have now skipped building the image if the MongoDB profile is disabled. Would you mind taking a look again?

File: tests/pom.xml [hunk 2]
```diff
@@ -228,6 +244,55 @@ Test cases are run against Docker images of Hono server + (Apache Qpid Dispatch
...
+                    <network>
+                      <mode>custom</mode>
+                      <name>hono</name>
+                      <alias>${hono.mongodb.host}</alias>
+                    </network>
+                    <memorySwap>314572800</memorySwap>
+                    <memory>314572800</memory>
+                    <log>
+                      <prefix>MONGODB</prefix>
+                      <color>green</color>
```
Comment (sophokles73) [inline tests/pom.xml:288]: please use `${log.color.hono-device-registry}` instead

Comment (kaniyan) [reply tests/pom.xml]: I have updated it.

**Pull request merged**